### PR TITLE
Init reed solomon once

### PR DESCRIPTION
**Pull request by ailisp**

### CODE REVIEW

File: chain/chunks/src/lib.rs [hunk 1]
```diff
@@ -511,12 +512,11 @@ impl ShardsManager {
 
     pub fn check_chunk_complete(
         data_parts: usize,
-        total_parts: usize,
         chunk: &mut EncodedShardChunk,
+        rs: &ReedSolomon<reed_solomon_erasure::galois_8::Field>,
```
Comment (SkidanovAlex) [inline chain/chunks/src/lib.rs]: Let's give this type some alias in `primitives`. Like `NightshadeReedSolomon` or something.

File: core/primitives/src/sharding.rs [hunk 2]
```diff
@@ -219,6 +215,7 @@ impl EncodedShardChunk {
         shard_id: ShardId,
         total_parts: usize,
         data_parts: usize,
```
Comment (SkidanovAlex) [inline core/primitives/src/sharding.rs]: Remove these two parameters and compute them from the `rs`.

File: chain/chunks/src/lib.rs [hunk 3]
```diff
@@ -6,6 +6,7 @@ use std::time::{Duration, Instant};
 
 use log::{debug, error};
 use rand::seq::SliceRandom;
+use reed_solomon_erasure::ReedSolomon;
```
Comment (mikhailOK) [inline chain/chunks/src/lib.rs]: We can do `use reed_solomon_erasure::galois_8::ReedSolomon` here

File: chain/chunks/src/lib.rs [hunk 4]
```diff
@@ -798,17 +793,17 @@ impl ShardsManager {
         outgoing_receipts_root: CryptoHash,
         tx_root: CryptoHash,
         signer: &dyn Signer,
+        rs: &ReedSolomon,
     ) -> Result<(EncodedShardChunk, Vec<MerklePath>), Error> {
-        let total_parts = self.runtime_adapter.num_total_parts();
-        let data_parts = self.runtime_adapter.num_data_parts();
+        let total_parts = rs.total_shard_count();
+        let data_parts = rs.data_shard_count();
```
Comment (SkidanovAlex) [inline chain/chunks/src/lib.rs]: are these ever used?

Comment (ailisp) [reply chain/chunks/src/lib.rs]: good  catch. no

**Pull request merged**